src/posit/connect/oauth/associations.py

```diff
@@ -0,0 +1,77 @@
+"""OAuth association resources."""
+
+from typing import List 
+
+from ..resources import Resource, ResourceParameters, Resources
+
+
+class Association(Resource):
+    pass
+
+
+class IntegrationAssociations(Resources):
+    """IntegrationAssociations resource."""
+
+    def __init__(self, params: ResourceParameters, integration_guid: str) -> None:
+        super().__init__(params)
+        self.integration_guid = integration_guid 
+
+    def find(self) -> List[Association]:
+        """Find OAuth associations.
+
+        Returns
+        -------
+        List[Association]
+        """
+        path = f"v1/oauth/integrations/{self.integration_guid}/associations"
+        url = self.params.url + path
+
+        response = self.params.session.get(url)
+        return [
+            Association(
+                self.params,
+                **result,
+            )
+            for result in response.json()
+        ]
+
+
+class ContentItemAssociations(Resources):
+    """ContentItemAssociations resource."""
+
+    def __init__(self, params: ResourceParameters, content_guid: str) -> None:
+        super().__init__(params) 
+        self.content_guid = content_guid
+
+
+    def find(self) -> List[Association]:
+        """Find OAuth associations.
+
+        Returns
+        -------
+        List[Association]
+        """
+        path = f"v1/content/{self.content_guid}/oauth/integrations/associations"
+        url = self.params.url + path
+        response = self.params.session.get(url)
+        return [
+            Association(
+                self.params,
+                **result,
+            )
+            for result in response.json()
+        ]
+
+
+    def update(self, integration_guid: None | str) -> None:
```

Comment:
@dbkegley @tdstein would love thoughts on the best way to represent this API interaction through the SDK. I don't think this is precisely what we want, but its a little ambiguous. 

The actual API endpoint accepts a list of json objects - this is designed with future flexibility in mind, but for now, the only valid inputs are: 

[{"oauth_integration_guid": <guid>}] to associate a content item with an integration
[] to dissociate the content item with all integrations

Given that currently you can only set one association per content item, I think it makes sense to support a function signature that accepts a single integration guid outside of a list. This breaks the semantics of PUTting an empty list to dissociate the content from integrations, though. 

I put `None` in as a placeholder for now to facilitate discussion, but I don't really like it all that much. I'm not sure what a better alternative would be.

Reply:
It feels like the correct expression is `content.associations.create(integration_guid)`, which should return a simple object that maintains the integration_guid and provides a `delete` method. The delete method would need to remove the association from the list.

```
class Association(Resource):
    def delete(self) -> None:
        # call `v1/content/{self.content_guid}/oauth/integrations/association` to remove the association.
```

fwiw, here is how I am thinking about the REST API implementation in my head, even though I know this doesn't represent the actual implementation.

Create an association
```
POST .../oauth/integrations/associations
{
    'integration_guid': 'whatever..."
}

Response:
{
    'guid': 'the-association-guid',
    'integration_guid': 'whatever...',
    'content_guid': 'the-content-guid'
}
```

Delete an association
```
DELETE .../the-content-guid/oauth/integrations/associations/the-association-guid
```

Reply:
in the current API implementation, you can't create / delete singular oauth associations - there aren't public CRUD operations that let you use an association in isolation. So I don't think your proposed pattern would be possible. 

You can only PUT a list of oauth integration guids to the`<content_guid>oauth/integrations/associations` endpoint - any PUTs completely wipe the current set of associations and replace it with a new set of associations.

Reply:
I see. When support for multiple associations is added, will the PUT operation still be an overwrite? If so, how should multiple people coordinate the modification when they each want to add a single association to the list simultaneously?

Reply:
its a good question. I don't know if we have an answer to this - currently, the asnwer is "whoever PUTs last wins." 

Really it should only be publishers / admins who use this endpoint to interact with content that they own, so I wouldn't expect there to be many use cases where there's parallel requests coming from many clients. We really built the API with the expected client primarily being the dashboard.